File: FactoryDesignPattern/Vehicle_Library_Example/SOL2_FactoryDP/Driver.java
package FactoryDesignPattern.Vehicle_Library_Example.SOL2_FactoryDP;

public class Driver {
  public static void main(String[] agrs) {
    VehicleFactory carFactory = new CarFactory();
    VehicleFactory bikeFactory = new BikeFactory();

    Client client1 = new Client(carFactory);
    Vehicle veh1 = client1.getVehicle();
    System.out.println("Client1 got veh1:");
    veh1.printVehicle();

    Client client2 = new Client(bikeFactory);
    Vehicle veh2 = client2.getVehicle();
    System.out.println("Client2 got veh2:");
    veh2.printVehicle();
  }
}

File: FactoryDesignPattern/Vehicle_Libraby_Example_Python/Driver.py
from Client import Client
from Vehicles.Vehicle import Vehicle
from VehiclesFactory.CarFactory import CarFactory
from VehiclesFactory.BikeFactory import BikeFactory
from VehiclesFactory.TruckFactory import TruckFactory

class Driver():

  @staticmethod
  def main():
    client1: Client = Client(CarFactory())
    client2: Client = Client(BikeFactory())
    client3: Client = Client(TruckFactory())

    veh1CreatedByClient1: Vehicle = client1.getCreatedVehicle()
    veh2CreatedByClient2: Vehicle = client2.getCreatedVehicle()
    veh3CreatedByClient3: Vehicle = client3.getCreatedVehicle()

    veh1CreatedByClient1.describe()
    veh2CreatedByClient2.describe()
    veh3CreatedByClient3.describe()

    print(veh1CreatedByClient1.getregistrationInfo())
    print(veh2CreatedByClient2.getregistrationInfo())
    print(veh3CreatedByClient3.getregistrationInfo())

if __name__ == "__main__":
  Driver.main()